src/Jackett/Services/IndexerManagerService.cs

```diff
@@ -194,6 +194,8 @@ private string GetIndexerConfigFilePath(IIndexer indexer)
 
         public void SaveConfig(IIndexer indexer, JToken obj)
         {
+            indexer.LoadFromSavedConfiguration(obj); // This seems crazy, but without this call the indexer will represent itself as unconfigured (which is misleading).
```

Comment:
All. The problem is that any indexer you add will say that it's unconfigured when IndexerManager saves it, the indexer will set the flag afterwards. However since IndexerManagerService owns the AggregateIndexer, it must update the list of configured indexers any time.
The problem is that BaseIndexer sets the IsConfigured flag after it calls SaveConfig on IndexerManagerService, so when it wants to update the AggregateIndexer, the filter for IsConfigured being true doesn't return the newly added indexer.
Since this is essentially configuring the indexer with the data it's going to write to the config file, this call seems harmless (if it wouldn't be, it would mean that instantly killing and restarting Jackett after adding an indexer would not be working).
If there's a better way, please let me know, I'm willing to change. However this seemed like the easiest one without going into bigger refactoring upon who does what in what order and who owns who.